# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# Lane - Refresh / Content Opportunity Scoring

I chose this lane because it addresses a practical problem faced by content teams: deciding which webpages should be reviewed first when time and resources are limited. For example, a  website may have 30,000 pages but only 3 writers. How will they decide which pages should be given the attention first? Instead of manually inspecting thousands of pages one by one, We can take the help of machine learning which will analyze multiple observable search and engagement signals to help reviewers know which pages should be given attention first.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Research Question
Which webpages should be reviewed first for content refresh based on observable search and engagement signals?

### Decision
My work helps someone decide which webpages should be prioritized for manual review.

### Who acts on it?
My recommendations can be used by Content writers, SEO specialists, or content managers etc.

### Action
- Refresh outdated content
- Expand the content
- Improve titles and metadata
- Monitor pages instead of changing them

### Cost of a wrong recommendation
- Time and effort will be wasted reviewing a page that didn't need attention.
- Some important page might continue to decline if it is not recommended for review, potentially reducing the traffic and missed opportunities.

### Unit of Analysis
Each row in the dataset represents a single webpage, so the unit of analysis is an individual webpage.

### Output
The output will be a ranked list of webpages that should be prioritized for manual review.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print(f"Rows: {len(df)}")
print(f"Columns: {len(df.columns)}")

corr = df["search_volume"].corr(df["impressions_90d"])
print(f"Correlation between search_volume and impressions_90d: {corr:.4f}")

Rows: 30000
Columns: 44
Correlation between search_volume and impressions_90d: 0.0012


The starter dataset contains **30,000 webpages** with **44 features**, providing enough information to explore patterns related to search performance. I also observed that the correlation between `search_volume` and `impressions_90d` is **0.0012**, which suggests that search volume alone does not explain page impressions. This indicates that multiple signals should be considered when prioritizing webpages for manual review, making the Refresh / Content Opportunity Scoring lane suitable for machine learning.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project is intended to support decision-making rather than replace human judgment. The recommendations identify webpages that may deserve manual review based on patterns in the available data, but they do not guarantee that refreshing a page/reviewing a page will improve its performance. The model identifies relationships in the data, not cause-and-effect, so any recommendation should be reviewed by content or SEO teams before taking action. But along with this it cannot predict Google's ranking algorithm or guarantee future search performance.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.